datahub-web-react/src/app/tags/CreateNewTagModal/index.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useState } from 'react';
+import { message } from 'antd';
+import { Modal } from '@components';
+import { useCreateTagMutation } from '../../../graphql/tag.generated';
+import { useEnterKeyListener } from '../../shared/useEnterKeyListener';
+import { useUserContext } from '../../context/useUserContext';
+import { useBatchAddOwnersMutation, useSetTagColorMutation } from '../../../graphql/mutations.generated';
+import { CreateNewTagModalProps, ModalButton, PendingOwner } from './types';
+import TagDetailsSection from './TagDetailsSection';
+import OwnersSection from './OwnersSection';
+
+/**
+ * Modal for creating a new tag with owners and applying it to entities
+ */
+const CreateNewTagModal: React.FC<CreateNewTagModalProps> = ({ onClose, open }) => {
+    // Tag details state
+    const [tagName, setTagName] = useState('');
+    const [tagDescription, setTagDescription] = useState('');
+    const [tagColor, setTagColor] = useState('#1890ff');
+
+    // Owners state
+    const [selectedOwnerUrns, setSelectedOwnerUrns] = useState<string[]>([]);
+    const [pendingOwners, setPendingOwners] = useState<PendingOwner[]>([]);
+
+    // Loading state
+    const [isLoading, setIsLoading] = useState(false);
+
+    // Mutations
+    const [createTagMutation] = useCreateTagMutation();
+    const [setTagColorMutation] = useSetTagColorMutation();
+    const [batchAddOwnersMutation] = useBatchAddOwnersMutation();
+
+    // Check permissions using UserContext
+    const userContext = useUserContext();
+    const canCreateTags = userContext?.platformPrivileges?.createTags || userContext?.platformPrivileges?.manageTags;
+
+    /**
+     * Handler for creating the tag and applying it to entities
+     */
+    const onOk = async () => {
+        if (!canCreateTags) {
+            message.error('You do not have permission to create tags');
+            return;
+        }
+
+        if (!tagName) {
+            message.error('Tag name is required');
+            return;
+        }
+
+        setIsLoading(true);
+
+        try {
+            // Step 1: Create the new tag
+            const createTagResult = await createTagMutation({
+                variables: {
+                    input: {
+                        id: tagName,
+                        name: tagName,
+                        description: tagDescription,
+                    },
+                },
+            });
+
+            const newTagUrn = createTagResult.data?.createTag;
+
+            if (!newTagUrn) {
+                throw new Error('Failed to create tag: No URN returned');
```

Comment:
Don't throw this error. 

Just message.error("Failed to create tag. An unexpected error occurred").

Try to never throw exceptions in the code, it will cause the user experience to completely bomb out.